### PR TITLE
HitTestResult.path should be unmodifiable

### LINKED ISSUE #4428: HitTestResult.path should be immutable
It's too tempting for people to try to manipulate it when they should instead just be putting things into the results list in the right order.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #4428

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/gestures/hit_test.dart [hunk 1]
```diff
@@ -47,24 +47,25 @@ class HitTestResult {
...
+    : _path = path ?? <HitTestEntry>[];
 
-  /// The list of [HitTestEntry] objects recorded during the hit test.
+  /// An unmodifiable list of [HitTestEntry] objects recorded during the hit test.
   ///
   /// The first entry in the path is the most specific, typically the one at
   /// the leaf of tree being hit tested. Event propagation starts with the most
   /// specific (i.e., first) entry and proceeds in order through the path.
-  final List<HitTestEntry> path;
+  List<HitTestEntry> get path => new List<HitTestEntry>.unmodifiable(_path);
```
Comment (Hixie) [inline packages/flutter/lib/src/gestures/hit_test.dart]: is there any reason to expose it at all?

**Comment by Hixie** [comment]
LGTM, but I think I'd prefer the takePath model unless there's a good use case for seeing the path as it's being built.

File: packages/flutter/lib/src/gestures/hit_test.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/gestures/hit_test.dart]: It's used by the testing harness to do non-trivial things.  We could expose an iterator instead, but there's no reason not to allow random access.

**Comment by abarth** [comment]
WidgetTester.dispatchEvent wouldn't work with the `takePath` approach.  It assumes you can dispatch to a given HitTestResult multiple times.

Actually, the whole pointer event capturing model wouldn't work with takePath because we capture the whole hit test result and dispatch using its path repeatedly.

**Comment by Hixie** [comment]
Ah, yeah. Ok, LGTM.

**Pull request merged**